src/tenant_tokens.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+    errors::* 
+};
+use serde::{Serialize, Deserialize};
+use jsonwebtoken::{encode, Header, EncodingKey};
+use time::{OffsetDateTime};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")] 
+struct TenantTokenClaim {
+    api_key_prefix: String,
+    search_rules: serde_json::Value,
+    #[serde(with = "time::serde::timestamp::option")]
+    exp: Option<OffsetDateTime>,
+}
+
+pub fn generate_tenant_token(search_rules: serde_json::Value, api_key: &str, expires_at: Option<OffsetDateTime>) -> Result<String, Error> {
+    if api_key.len() < 8 {
+        return Err(Error::TenantTokensInvalidApiKey)
+    }
+
+    if expires_at.is_some() && OffsetDateTime::now_utc() > expires_at.unwrap() {
+        return Err(Error::TenantTokensExpiredSignature)
+    }
+
+    let key_prefix = &api_key[..8];
+    let claims = TenantTokenClaim {
+        api_key_prefix: key_prefix.to_string(),
+        exp: expires_at,
+        search_rules
+    };
+
+    let token = encode(
+        &Header::default(),
+        &claims,
+        &EncodingKey::from_secret(api_key.as_bytes()),
+    );
+
+    Ok(token.unwrap())
```

Comment:
Is this `unwrap` safe?
Shouldn't we return an error instead of panicking if it fails?

Reply:
Solved in earlier versions :)